BDArmory/Weapons/Missiles/MissileBase.cs

```diff
@@ -352,24 +363,47 @@ public void FloatingOriginCorrection()
             }
         }
 
+        public ModuleMissileRearm reloadableRail;
```

Comment:
The compiler is giving
```
'MissileLauncher.reloadableRail' hides inherited member 'MissileBase.reloadableRail'. Use the new keyword if hiding was intended. [BDArmory]
```

I think the best way to resolve this is to remove `reloadableRail` from `MissileLauncher` so that it uses the one in `MissileBase`. However, there's then an issue of the order of initialisation with the way you've written it below since `MissileLauncher.OnStart` adds the MMR module dynamically (`Start` is called before `OnStart` I think). If you move the lines 379—386 back into `AmmoCount.get` as originally written and `AmmoCount` is only called after the MMR module has been added to the part, then it should work fine.
Also, setting `hasAmmo`'s access level to `protected` will allow resetting the ammo check for any modules that add the MMR module dynamically by setting `hasAmmo = true`.

Reply:
The MMR getting dynamically added is a sanity check in case a clustermissile/multimissile launcher's .cfg is improperly setup, so things will still (mostly) work.
It is not getting added under standard circumstances, or if a standard missile, which is why I moved the `findModule...` bit to `Start()`, since that way it ensures the module (if there from the start) gets found and is available, rather than being reliant on a future `GetMissileCount()` call (which should be getting called before a missile has a chance to fire, but...)

Reply:
OK, fair enough. Yeah, as long as you're setting `hasAmmo = true` after adding the MMR in the sanity check, then that's fine.